hasktorch/src/Torch/NN.hs

```diff
@@ -139,3 +139,35 @@ instance Parameterized Linear
 --     return $ Linear{..}
 
 instance Parameterized [Linear]
+
+data Conv2dSpec = 
+  Conv2dSpec {
+    inputChannelSize  :: Int, 
+    outputChannelSize :: Int, 
+    kernelSize0       :: Int, 
+    kernelSize1       :: Int
+    } deriving (Show, Eq)
+
+data Conv2d = 
+  Conv2d { 
+    conv2dWeight :: Parameter, 
+    conv2dBias   :: Parameter
+    } deriving (Show, Generic)
+
+conv2d'' :: Conv2d -> (Int, Int) -> (Int, Int) -> Tensor -> Tensor
+conv2d'' layer stride padding input = 
```

Comment:
Why does this have two apostrophes?

Reply:
The `Torch` module exports both `Torch.NN` and `Torch.Functional` modules and there are already definitions for `conv2d` and `conv2d'` present in `Torch.Functional` module hence the two apostrophes.

Reply:
We haven't formalized it quite yet, but a convention we've started using on functions which implement the 'forward' inference on data with paramaterized instances is "xxxForward". For example, have been doing that in this PR with RNNs https://github.com/hasktorch/hasktorch/pull/319/files

Reply:
I have made the necessary changes. Could you please have a look?